### PR TITLE
Add serialization of OperationsProblem and Simulation

### PR DESCRIPTION
**Pull request by daniel-thom**

### CODE REVIEW

File: src/core/simulation.jl [hunk 1]
```diff
@@ -688,3 +691,145 @@ function execute!(sim::Simulation; kwargs...)
...
+they passed to the original Simulation.
+
+# Arguments
+- `directory::AbstractString`: the directory returned from the call to serialize
+- `stage_info::Dict`: Two-level dictionary containing stage parameters that cannot be
+  serialized. The outer dict should be keyed by the stage name. The inner dict must contain
+  'optimizer' and may contain 'jump_model'. These should be the same values used for the
+  original simulation.
+"""
+function Simulation(directory::AbstractString, stage_info::Dict; kwargs...)
```
Comment (daniel-thom) [inline src/core/simulation.jl]: Accepting kwargs here is not idea.  I suggest we make a change to the Simulation constructor.
- We could make initial_times an optional parameter to it, defaulting to nothing.
- Remove kwargs from Simulation().
- Remove kwargs this constructor.  initial_times will have been serialized, so we will get it back. There won't be any ambiguity.

File: test/test_simulation_execute.jl [hunk 2]
```diff
@@ -78,190 +88,209 @@ function test_load_simulation(file_path::String)
...
+    for i in 1:2
+        if i == 1
+            build!(sim)
+        else
+            rm(g_test_path, recursive = true)
+            mkdir(g_test_path)
+            sim = get_deserialized(sim, stage_info)
         end
-    end
+        sim_results = execute!(sim; constraints_duals = duals)
```
Comment (jd-lara) [inline test/test_simulation_execute.jl]: Execute shouldn't take constraints_duals anymore

File: src/core/simulation.jl (hunk 1 above)
Comment (jd-lara) [reply src/core/simulation.jl]: Ok, we can address this after merging. I will make initial_time an optional parameter.

Comment (daniel-thom) [reply src/core/simulation.jl]: Fixed it here.

**Pull request merged**